packages/presentation/lib/screen/home/home_page.dart

```diff
@@ -1,177 +1,159 @@
-          titleText = 'You lose!';
-          contentText = 'The attempts are over. Try again!';
-          buttonText = 'Ok :(';
-        } else if (state.gameState == GameState.win) {
-          titleText = 'You won!';
-          contentText = 'The number was : ${state.generatedNumber}';
-          buttonText = 'Generate new number';
-        }
-
-        showDialog(
+    bloc.dialogStream.listen(
+      (event) {
+        if (event is GameAlertDialog) {
+          showDialog(
             context: context,
             barrierDismissible: false,
             builder: (context) {
               return ShowDialogWidget(
-                  titleText: titleText,
-                  contentText: contentText,
-                  buttonText: buttonText,
+                  titleText: event.titleText,
+                  contentText: event.contentText,
+                  buttonText: event.buttonText,
                   widgetFunction: () {
-                    BlocProvider.of<HomeBloc>(context).add(
-                      GenerateNumberEvent(),
-                    );
+                    bloc.generate();
                     Navigator.of(context).pop();
                   });
-            });
+            },
+          );
+        }
       },
-      child: Scaffold(
-        appBar: AppBar(
-          title: Text(widget.title),
-        ),
-        body: Center(
-          child: Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 15.0),
-            child: Column(
-              mainAxisAlignment: MainAxisAlignment.center,
-              children: <Widget>[
-                BlocSelector<HomeBloc, HomeState, int?>(
-                  selector: (state) => state.generatedNumber,
-                  builder: (context, generatedNumber) {
-                    return generatedNumber != null
-                        ? Text('Random number: $generatedNumber')
-                        : const SizedBox();
-                  },
-                ),
-                const Text(
-                  'Try to predict generated number:',
-                  style: TextStyle(fontSize: 20),
-                ),
-                const SizedBox(height: 15),
-                TextFormField(
-                  keyboardType: TextInputType.number,
-                  maxLength: 1,
-                  controller: _controller,
-                  decoration: const InputDecoration(
-                    border: OutlineInputBorder(),
-                    labelText: 'Number?',
-                  ),
-                  onChanged: (value) {
-                    BlocProvider.of<HomeBloc>(context).add(
-                      ChangeSubmitButtonStateEvent(
-                          isSubmitButtonActive: value.isNotEmpty),
-                    );
-                  },
-                ),
-                const SizedBox(height: 15),
-                Row(
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
```

Comment:
you can move your controller in bloc and make method dispose in abstract class HomeBloc, then, in BlocScreenState call dispose and call bloc.dispose where will you dispose controller.

Reply:
I just remove the controller